strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-nuget-layout-provider/src/main/java/org/carlspring/strongbox/repository/NugetRepositoryFeatures.java

```diff
@@ -1,10 +1,112 @@
 package org.carlspring.strongbox.repository;
 
+import java.io.IOException;
+import java.net.URISyntaxException;
+import java.util.HashSet;
+import java.util.Set;
+
+import javax.inject.Inject;
+
+import org.carlspring.strongbox.artifact.coordinates.NugetHierarchicalArtifactCoordinates;
+import org.carlspring.strongbox.client.ArtifactTransportException;
+import org.carlspring.strongbox.configuration.Configuration;
+import org.carlspring.strongbox.configuration.ConfigurationManager;
+import org.carlspring.strongbox.domain.RemoteArtifactEntry;
+import org.carlspring.strongbox.services.ArtifactEntryService;
+import org.carlspring.strongbox.storage.Storage;
+import org.carlspring.strongbox.storage.repository.Repository;
+import org.carlspring.strongbox.storage.repository.remote.RemoteRepository;
+import org.springframework.stereotype.Component;
+
+import ru.aristar.jnuget.client.NugetClient;
+import ru.aristar.jnuget.rss.PackageEntry;
+import ru.aristar.jnuget.rss.PackageFeed;
+
 /**
+ * 
  * @author carlspring
+ * @author Sergey Bespalov
  */
+@Component
 public class NugetRepositoryFeatures
         implements RepositoryFeatures
 {
 
+    @Inject
+    private ConfigurationManager configurationManager;
+
+    @Inject
+    private ArtifactEntryService artifactEntryService;
+
+    @Override
+    public void downloadRemoteIndex(String storageId,
+                                    String repositoryId)
+        throws ArtifactTransportException,
+        RepositoryInitializationException
+    {
+        Storage storage = getConfiguration().getStorage(storageId);
+        Repository repository = storage.getRepository(repositoryId);
+        RemoteRepository remoteRepository = repository.getRemoteRepository();
+        if (remoteRepository == null)
+        {
+            return;
+        }
+        String remoteRepositoryUrl = remoteRepository.getUrl();
+
+        try (NugetClient nugetClient = new NugetClient())
+        {
+            nugetClient.setUrl(remoteRepositoryUrl);
+            int packageCount;
+            try
+            {
+                packageCount = nugetClient.getPackageCount(false);
+            }
+            catch (IOException | URISyntaxException e)
+            {
+                throw new ArtifactTransportException(e);
+            }
+            for (int i = 0; i < packageCount / 100 + packageCount % 100; i++)
```

Comment:
Could you please explain (for my better understanding) what this loop is about?

Reply:
it's the iteration for Feed pages